### PR TITLE
Add Policy Based Design Pattern demo and some info about it.

### PR DESCRIPTION
**Pull request by n-is**

### CODE REVIEW

File: code/design_pattern/src/policy_based_design/policy_design.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#include <iostream>
+
+class Classical
+{
+public:
+        Classical(int a) { }
+
+        void solve() {
+                std::cout << "The differential equation was solved by the "
+                                "classical method." << std::endl;
```
Comment (arnavb) [inline code/design_pattern/src/policy_based_design/policy_design.cpp]: Don't use so much indentation; Use for spaces for each level.

File: code/design_pattern/src/policy_based_design/policy_design.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        }
+};
+
+class Laplace
+{
+public:
+        void solve() {
+                std::cout << "The differential equation was solved by the "
+                                "Laplace Transform method." << std::endl;
```
Comment (arnavb) [inline code/design_pattern/src/policy_based_design/policy_design.cpp]: Same here.

File: code/design_pattern/src/policy_based_design/readme.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Policy Based Design
```
Comment (arnavb) [inline code/design_pattern/src/policy_based_design/readme.md]: Follow our [documentation guide](https://github.com/OpenGenus/cosmos/blob/master/guides/documentation_guide.md), please.

**Comment by n-is** [comment]
@arnavb what do you mean by empty implementation? I don't understand.

File: code/design_pattern/src/policy_based_design/readme.md (hunk 3 above)
Comment (n-is) [reply code/design_pattern/src/policy_based_design/readme.md]: Do I really need to search for images and do everything written in that guide?

**Comment by n-is** [comment]
That is what is there to understand about design patterns right. It is a setup for the code.

File: code/design_pattern/src/policy_based_design/readme.md (hunk 3 above)
Comment (arnavb) [reply code/design_pattern/src/policy_based_design/readme.md]: @n-is Well an image isn't necessary (You can replace with an explanation), but the rest of it is. We have a agreed-upon style, please follow it!

Comment (n-is) [reply code/design_pattern/src/policy_based_design/readme.md]: @arnavb Thanks for pointing that out. I have followed the guide and removed first person.

**Comment by n-is** [comment]
@arnavb Well, I learned how to do, maybe just a little. And so that was not inconvenience. Thanks.
And one question, Why is it showing "All checks have failed". I don't know about travis.

**Comment by arnavb** [comment]
@n-is Our Travis CI is currently in beta, so if you have C _or_ C++ code, then you should check if your file doesn't compile. Once you fix it (If there even is an error), then move on! There are many errors we have to fix in our Travis build currently, but once its done, then it will be more reliable! Kudos!

**Comment by n-is** [comment]
@arnavb Then, why is there still error?

**Comment by arnavb** [comment]
@n-is Like I said, our Travis build is in beta right now. 😆 Thus, there are many _other_ files that have errors, not yours.

**Comment by n-is** [comment]
Oh, now I understand.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**